crates/primitives/src/revm/env.rs

```diff
@@ -159,63 +184,62 @@ pub fn fill_tx_env_with_beacon_root_contract_call(env: &mut Env, parent_beacon_b
 }
 
 /// Fill transaction environment from [TransactionSignedEcRecovered].
-pub fn fill_tx_env_with_recovered(tx_env: &mut TxEnv, transaction: &TransactionSignedEcRecovered) {
-    fill_tx_env(tx_env, transaction.as_ref(), transaction.signer())
+pub fn fill_tx_env_with_recovered(
+    tx_env: &mut TxEnv,
+    transaction: &TransactionSignedEcRecovered,
+    #[cfg(feature = "optimism")] envelope: Bytes,
+) {
+    #[cfg(not(feature = "optimism"))]
+    fill_tx_env(tx_env, transaction.as_ref(), transaction.signer());
+
+    #[cfg(feature = "optimism")]
+    fill_tx_env(tx_env, transaction.as_ref(), transaction.signer(), envelope);
 }
```

Comment:
this function should be duplicated instead

Reply:
addressed in https://github.com/anton-rs/op-reth/pull/135